hs_core/views/resource_folder_hierarchy.py

```diff
@@ -83,16 +82,22 @@ def data_store_structure(request):
             logical_file_type = ''
             logical_file_id = ''
             for f in ResourceFile.objects.filter(object_id=resource.id):
-                if name_with_rel_path == get_resource_file_name_and_extension(f)[0]:
+                if name_with_full_path == f.storage_path:
                     f_pk = f.pk
                     f_url = get_resource_file_url(f)
                     if resource.resource_type == "CompositeResource":
                         logical_file_type = f.logical_file_type_name
                         logical_file_id = f.logical_file.id
                     break
 
-            files.append({'name': fname, 'size': size, 'type': mtype, 'pk': f_pk, 'url': f_url,
-                          'logical_type': logical_file_type, 'logical_file_id': logical_file_id})
+            if f_pk != '':
+                files.append({'name': fname, 'size': size, 'type': mtype, 'pk': f_pk, 'url': f_url,
+                              'logical_type': logical_file_type,
+                              'logical_file_id': logical_file_id})
+            else:
+                logger.error("data_store_structure: filename {} in iRODs has no analogue in Django"
+                             .format(name_with_full_path))
+            # TODO: flag when files in Django have no analogue in iRODs

```

Comment:
@hyi This is the only knowledge that the method has. The other way requires a comprehensive scan the way that `check_irods_files` does. This is incomplete knowledge, but important to report, because

1. The reporting does not slow down the code and 
2. There is no extra code beyond accomplishing the task used in inferring it.